front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -392,3 +415,23 @@ export function FixedAssistantInputBar({
     </div>
   );
 }
+
+function isMarkdownFile(file: File): boolean {
+  if (file.type === "") {
+    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+    // Check if the file extension corresponds to a markdown file
+    return fileExtension === "md" || fileExtension === "markdown";
+  }
+  return file.type === "text/markdown";
+}
+
+function copyFileWithMimeType(originalFile: File, mimeType: string): File {
```

Comment:
Let's avoid doing some kind of copy of "file". We use APIs to create a file, let's not mess with this. Instead feel free to create a helper something like `getMimeTypeFromFile()` and use it when you want to access the `mimeType`, WDYT?